src/commands/publish/mod.rs

```diff
@@ -107,21 +95,53 @@ fn publish_script(
         failure::bail!(error_msg(res_status, res_text))
     }
 
-    let pattern = if target.route.is_some() {
-        log::info!("publishing to route");
-        publish_route(&user, &target)?
-    } else {
+    let routes = target.routes()?;
+    log::info!("routes: {:#?}", &routes);
+
+    if routes.is_empty() {
+        // this is a zoneless deploy
         log::info!("publishing to subdomain");
-        publish_to_subdomain(target, user)?
-    };
+        let deploy_address = publish_to_subdomain(target, user)?;
 
-    log::info!("{}", &pattern);
-    message::success(&format!(
-        "Successfully published your script to {}",
-        &pattern
-    ));
+        message::success(&format!(
+            "Successfully published your script to {}",
+            deploy_address
+        ));
 
-    Ok(())
+        Ok(())
+    } else {
+        // this is a zoned deploy
+        log::info!("publishing to zone");
+        // Most users of Workers Sites will be confused by our route patterns; this warning
+        // informs them of the best configuration for their project, but does not block them
+        // from deploying to a non-wildcard route.
+        if target.site.is_some() && routes.len() == 1 {
+            let route_pattern = &routes[0].pattern;
+            if !route_pattern.ends_with("*") {
+                message::warn(&format!("The route in your wrangler.toml should have a trailing * to apply the Worker on every path, otherwise your site will not behave as expected.\nroute = {}*", route_pattern));
+            }
+        }
+
+        if let Some(zone_id) = &target.zone_id {
+            let published_routes = publish_routes(&user, routes, zone_id)?;
+
+            message::success(&format!(
+                "Successfully published your script to:\n{}",
+                format_routes(published_routes)
+            ));
+
+            Ok(())
+        } else {
+            failure::bail!("You must provide a zone_id in your wrangler.toml before publishing!")
```

Comment:
```suggestion
            failure::bail!("You must provide a zone_id in your wrangler.toml before publishing to a route!")
```

Maybe worth putting something in here about the workers.dev alternative as well - not sure

Reply:
i think i may be addressing more of the workers.dev stuff in subsequent work; as it is this is just the same message we had before but idk why i changed the wording.